packages/runtime/test/transport/identityRecoveryKits.test.ts

```diff
@@ -0,0 +1,124 @@
+import { OwnerRestriction } from "../../src";
+import { RuntimeServiceProvider, TestRuntimeServices } from "../lib";
+
+const serviceProvider = new RuntimeServiceProvider();
+
+let services: TestRuntimeServices;
+let servicesWithDisabledDatawallet: TestRuntimeServices;
+
+beforeAll(async () => {
+    const runtimeServices = await serviceProvider.launch(1, { enableDatawallet: true });
+    services = runtimeServices[0];
+
+    const datawalletDisabledServices = await serviceProvider.launch(1, { enableDatawallet: false });
+    servicesWithDisabledDatawallet = datawalletDisabledServices[0];
+}, 30000);
+
+afterAll(async () => await serviceProvider.stop());
+
+afterEach(async () => {
+    const devicesController = services.transport.identityRecoveryKits["createIdentityRecoveryKitUseCase"]["devicesController"];
+    const tokenController = services.transport.identityRecoveryKits["createIdentityRecoveryKitUseCase"]["tokenController"];
+
+    const devices = await devicesController.list();
+
+    const backupDevices = devices.filter((device) => device.isBackupDevice);
+    for (const backupDevice of backupDevices) {
+        const matchingTokens = await tokenController.getTokens({
+            "cache.content.@type": "TokenContentDeviceSharedSecret",
+            "cache.content.sharedSecret.id": backupDevice.id.toString()
+        });
+
+        for (const matchingToken of matchingTokens) {
+            await tokenController.delete(matchingToken);
+        }
+
+        await devicesController.delete(backupDevice);
+    }
+});
+
+describe("Identity Recovery Kits", () => {
+    test("should create a recovery kit", async () => {
+        const response = await services.transport.identityRecoveryKits.createIdentityRecoveryKit({
+            profileName: "profileName",
+            passwordProtection: { password: "aPassword" }
+        });
+        expect(response).toBeSuccessful();
+
+        const devices = (await services.transport.devices.getDevices()).value;
+        const backupDevices = devices.filter((device) => device.isBackupDevice);
+        expect(backupDevices).toHaveLength(1);
+
+        const backupDevice = backupDevices[0];
+
+        const tokens = (await services.transport.tokens.getTokens({ ownerRestriction: OwnerRestriction.Own, query: { expiresAt: "^9999-12-31" } })).value;
```

Comment:
Is the `^` in the expiry date there on purpose?

Reply:
yes, regex matcher